giskard/project.py

```diff
@@ -186,6 +191,17 @@ def upload_df(
                              )
         return result
 
+    def _validate_data(self, column_types, df, target):
```

Comment:
since this method doesn't only do the validation, I'd call it "_prepare_data" or "_validate_and_compress_data"
Otherwise it's not clear why it returns results

Reply:
"_validate_and_compress_data" sounds good for me . I ll update it .